src/tasks/component-installers/devfile-workspace-operator-installer.ts

```diff
@@ -25,26 +29,54 @@ export class DevWorkspaceTasks {
   protected kubeHelper: KubeHelper
   protected cheHelper: CheHelper
   protected openShiftHelper: OpenShiftHelper
+  protected certManagerTask: CertManagerTasks
 
-  protected devWorkspaceServiceAccount = 'devworkspace-controller'
-  protected devWorkspaceRole = 'devworkspace-controller'
+  protected devWorkspaceServiceAccount = 'devworkspace-controller-serviceaccount'
 
-  protected devWorkspaceEditWorkspaceClusterRole = 'edit-workspaces'
-  protected devWorkspaceViewWorkspaceClusterRole = 'view-workspaces'
-  protected devWorkspaceRoleBinding = 'devworkspace-controller'
+  // Roles
+  protected devWorkspaceLeaderElectionRole = 'devworkspace-controller-leader-election-role'
 
-  protected deploymentName = 'devworkspace-controller'
+  // Cluster Roles
+  protected devWorkspaceEditWorkspaceClusterRole = 'devworkspace-controller-edit-workspaces'
+  protected devworkspaceProxyClusterRole = 'devworkspace-controller-proxy-role'
+  protected devworkspaceClusterRole = 'devworkspace-controller-role'
+  protected devWorkspaceViewWorkspaceClusterRole = 'devworkspace-controller-view-workspaces'
+  // Cluster Role created by devworkspace pod
+  protected devWorkspaceClusterRoleWebhook = 'devworkspace-webhook-server'
+
+  // RoleBindings and ClusterRole Bindings necessary devworkspace
+  protected devWorkspaceLeaderElectionRoleBinding = 'devworkspace-controller-leader-election-rolebinding'
+  protected devworkspaceProxyClusterRoleBinding = 'devworkspace-controller-proxy-rolebinding'
+  protected devWorkspaceRoleBinding = 'devworkspace-controller-rolebinding'
+
+  // Deployment names
+  protected deploymentName = 'devworkspace-controller-manager'
+
+  protected devworkspaceConfigMap = 'devworkspace-controller-configmap'
+
+  protected devworkspaceCertificate = 'devworkspace-controller-serving-cert'
+  protected devworkspaceCertIssuer = 'devworkspace-controller-selfsigned-issuer'
+
+  // DevWorkspace CRD Names
+  protected devworkspacesCRDName = 'devworkspaces.workspace.devfile.io'
+  protected componentsCRDName = 'components.controller.devfile.io'
+  protected devworkspacetemplatesCRDName = 'devworkspacetemplates.workspace.devfile.io'
+  protected workspaceroutingsCRDName = 'workspaceroutings.controller.devfile.io'
+
+  protected webhooksName = 'controller.devfile.io'
 
   constructor(private readonly flags: any) {
     this.kubeHelper = new KubeHelper(flags)
     this.cheHelper = new CheHelper(flags)
     this.openShiftHelper = new OpenShiftHelper()
-
+    this.certManagerTask = new CertManagerTasks({ flags })
   }
 
-  getTemplatePath() {
-    return path.join(this.flags.templates, 'devworkspace')
-
+  getTemplatePath(ctx: any) {
```

Comment:
Let's initialize templates in a constructor.
If `this.flags.templates` isn't set then the base directory should be getEmbeddedTemplatesDirectory() I guess